src/rt/array/capacity.d

```diff
@@ -0,0 +1,197 @@
+/**
+ This module contains support for controlling dynamic arrays' capacity
+
+  Copyright: Copyright Digital Mars 2000 - 2019.
+  License: Distributed under the
+       $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+     (See accompanying file LICENSE)
+  Source: $(DRUNTIMESRC rt/_array/_capacity.d)
+*/
+module rt.array.capacity;
+
+// HACK:  This is a lie.  `_d_arraysetcapacity` is neither `nothrow` nor `pure`, but this lie is
+// necessary for now to prevent breaking code.
+private extern (C) size_t _d_arraysetcapacity(const TypeInfo ti, size_t newcapacity, void[]* arrptr) pure nothrow;
```

Comment:
This important note was absent in `object.d`.  I added it here for full transparency.